### DISCUSSION TITLE
mac silicon build dependencies?

### DISCUSSION THREAD

**Discussion by kleeuccs** [gem5/gem5#380] [Q&A] [answered]
@BobbyRBruce 

For MAC Silcone gem5 build, what dependencies from brew need to be installed?  There isn't a build-essentials, but scons exists.  I have python 3.11.x under pyenv.  I also installed gcc.

~/srccode/github-repo/gem5 stable > scons --version                                                                 node system 01:42:55 PM
SCons by Steven Knight et al.:
	SCons: v4.5.2.120fd4f633e9ef3cafbc0fec35306d7555ffd1db, Tue, 21 Mar 2023 12:11:27 -0400, by bdbaddog on M1DOG2021
	SCons path: ['/opt/homebrew/Cellar/scons/4.5.2/libexec/lib/python3.11/site-packages/SCons']
Copyright (c) 2001 - 2023 The SCons Foundation

After each upgrade of OS, I update xcode utilities.

After installing scons and gcc from brew, I see:
~/srccode/github-repo/gem5 stable > python `which scons` build/X86/gem5.opt -j9                                 32s node system 12:49:07 PM
Traceback (most recent call last):
  File "/opt/homebrew/bin/scons", line 5, in <module>
    from SCons.Script.Main import main
ModuleNotFoundError: No module named 'SCons'

I didn't see any other scons modules that would help.

Any advice would be appreciated.

Thank you, 

Ken Lee

**Comment by BobbyRBruce** [comment]
As this was more of a question I moved it to the Q&A discussions section. I hope you don't mind.

**Comment by BobbyRBruce** [comment]
It's always worth getting this out the way first: have you tried deleting the build directory with `rm -rf build` and going with `scons build/X86/gem5.opt` first?

Changing your environment (like updating the OS) can mean the build file needs wiped and to build from scratch.

**Comment by kleeuccs** [comment, marked as answer]
I removed pyenv from .zshrc, and installed python 3.11.4 as the system version of python.  
For building on the mac M1, I get similar warnings and errors mentioned in #94.

ext/gdbremote/signals.hh:176:5: error: expected identifier
    EXC_BREAKPOINT = 150, //breakpoint
    ^
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/mach/exception_types.h:87:33: note: expanded from macro 'EXC_BREAKPOINT'
#define EXC_BREAKPOINT          6       /* Trace, breakpoint, etc. */
                                ^
 [     CXX] X86/python/_m5/param_BaseNonCachingSimpleCPU.cc -> .o
 [SO Param] m5.objects.BaseTimingSimpleCPU, BaseTimingSimpleCPU -> X86/python/_m5/param_BaseTimingSimpleCPU.cc
44 warnings and 6 errors generated.
scons: *** [build/X86/cpu/o3/probe/elastic_trace.o] Error 1
44 warnings generated.
scons: building terminated because of errors.
*** Summary of Warnings ***
Warning: You can get a 12% performance improvement by installing tcmalloc (libgoogle-perftools-dev package on Ubuntu or RedHat).
Warning: Info: Compatible header file <linux/kvm.h> not found, disabling KVM support.
Warning: Header file <png.h> not found.
         This host has no libpng library.
         Disabling support for PNG framebuffers.
Warning: Can't find library for POSIX clocks.
Warning: Couldn't find HDF5 C++ libraries. Disabling HDF5 support.
Warning: Cannot enable KVM, host seems to lack KVM support

I will try merging #209, and see if that helps as suggested in post #94.

Thank you,

Ken Lee

**Reply by BobbyRBruce** [reply]
Excellent! I'm 99% sure #209 will fix that bug and you'll be fine, but let me know just to be sure.
